x-pack/plugins/infra/server/plugin.ts

```diff
@@ -62,20 +62,17 @@ export const config: PluginConfigDescriptor<InfraConfig> = {
   schema: schema.object({
     enabled: schema.boolean({ defaultValue: true }),
     // Setting variants only allowed in the Serverless offering, otherwise always default `logs-ui` value
-    logs: schema.conditional(
-      schema.contextRef('serverless'),
-      true,
-      schema.object({
+    logs: offeringBasedSchema({
```

Comment:
`offeringBasedSchema` allows being explicit with the type as in:
```ts
logs: offeringBasedSchema<InfraConfig['logs']>(...)
```

I took a different approach, though... TS was thinking `string` because of the `options`. So I casted it `as const`.

Reply:
That makes sense, thanks a lot!